docs/using-the-compiler.rst

```diff
@@ -0,0 +1,271 @@
+          "*": {
+            "*": [ "metadata", "evm.bytecode" ]
+          },
+          // Enable the abi and opcodes output of MyContract defined in file def.
+          "def": {
+            "MyContract": [ "abi", "evm.opcodes" ]
+          },
+          // Enable the source map output of every single contract.
+          "*": {
+            "*": [ "evm.sourceMap" ]
+          },
+          // Enable the AST and Why output of every single file.
+          "*": {
+            "": [ "ast", "why3" ]
+          }
+        }
+      }
+    }
+
+
+Output Description
+------------------
+
+.. code-block:: none
+
+    {
+      // Optional: not present if no errors/warnings were encountered
+      errors: [
+        {
+          // Optional
+          file: "sourceFile.sol",
+          // Optional: Location within the source file.
+          sourceLocation: [
+            { start: 0, end: 100 },
+          ],
+          // Mandatory: Error type, such as "TypeError", "InternalCompilerError", "Exception", etc
+          type: "TypeError",
+          // Mandatory: Component where the error originated.
+          component: "general", "why3", "ewasm", etc.
+          // Mandatory
+          severity: "warning" or "error",
+          // Mandatory
+          message: "Invalid keyword"
+        }
+      ],
+      // This contains the file-level outputs. In can be limited/filtered by the outputSelection settings.
+      sources: {
+        "sourceFile.sol": {
+          // Identifier (used in source maps)
+          id: 1,
+          // The AST object
+          ast: {}
+        }
+      },
+      // This contains the contract-level outputs. It can be limited/filtered by the outputSelection settings.
+      contracts: {
+        "sourceFile.sol": {
+          "ContractName": {
+            // The Ethereum Contract ABI. If empty, it is represented as an empty array.
+            // See https://github.com/ethereum/wiki/wiki/Ethereum-Contract-ABI
+            abi: [],
+            evm: {
+              // Intermediate representation (string)
+              ir: "",
+              // Assembly (string)
+              assembly: "",
+              // Old-style assembly (string)
+              asm: "",
+              // Old-style assembly (JSON object)
+              asmJSON: [],
+              bytecode: {
+                // The bytecode as a hex string.
+                object: "00fe",
+                // The source mapping as a string. See the source mapping definition.
+                sourceMap: "",
+                // If given, this is an unlinked object (cannot be filtered out explicitly, might be
+                // filtered if both bytecode, runtimeBytecode, opcodes and others are filtered out)
+                linkReferences: {
+                  "libraryFile.sol": {
+                    // Byte offsets into the bytecode. Linking replaces the 20 bytes located there.
+                    "Library1": [
+                      { start: 0, end: 20 },
```

Comment:
Length sounds less confusing I think. Though we have the same start/end in the error location too.